File: src/Shared/Polyrific.Catapult.Shared.ApiClient/VersionService.cs
// Copyright (c) Polyrific, Inc 2018. All rights reserved.

using System.Threading.Tasks;
using Polyrific.Catapult.Shared.Dto.Version;
using Polyrific.Catapult.Shared.Service;

namespace Polyrific.Catapult.Shared.ApiClient
{
    public class VersionService : BaseService, IVersionService
    {
        public VersionService(IApiClient api) : base(api)
        {
        }

        public async Task<VersionDto> GetApiVersion()
        {
            return await Api.Get<VersionDto>("version");
        }
    }
}

File: src/Web/Polyrific.Catapult.Web/ClientApp/src/app/core/services/version.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { Observable } from 'rxjs';
import { VersionDto } from '../models/version/version-dto';

@Injectable()
export class VersionService {

  constructor(private api: ApiService) { }

  getVersions(): Observable<VersionDto> {
    return this.api.get<VersionDto>('version');
  }
}
